containers/src/Kokkos_Bitset.hpp

```diff
@@ -400,13 +400,7 @@ void deep_copy(Bitset<DstDevice>& dst, Bitset<SrcDevice> const& src) {
     Kokkos::Impl::throw_runtime_exception(
         "Error: Cannot deep_copy bitsets of different sizes!");
   }
```

Comment:
Should we remove these throws here? We have checks inside Kokkos::deep_copy for matching size. Though having it here would keep the more specific message.

Reply:
Not feeling too strongly about it.  I would merge the PR as is.

Reply:
it would have been in scope since the check clearly accompanies the impl deep copy call same as the fences